include/aspect/simulator.h

```diff
@@ -924,6 +924,30 @@ namespace aspect
        */
       void apply_limiter_to_dg_solutions (const AdvectionField &advection_field);
 
+
+      /**
+       * Compute the reactions in case of operator splitting:
+       * Using the current solution vector, this function makes a number of time
+       * steps determined by the size of the reaction time step, and solves a
+       * system of coupled ordinary differential equations for the reactions between
+       * compositional fields and temperature in each of them. To do that, is uses
+       * the reaction rates outputs from the material and heating models used in
+       * the computation. The solution vector is then updated with the new values
+       * of temperature and composition after the reactions.
+       *
+       * As the ordinary differential equation in any given point is independent
+       * from the solution at all other points, we do not have to assemble a matrix,
+       * but just need to loop over all points and compute the update to the solution.
+       *
+       * The function also updates the old solution with the reaction update so that
+       * the advection scheme will have the correct time stepping in the next step.
```

Comment:
No, I really mean the time stepping. Imagine the simple case where the velocity is zero, there is no diffusion, and the assembled matrix for the advection also does not include any reaction terms, because those are handled by the operator split. Then, in the first time step, all that is assembled is basically `scratch.phi_field[i] * scratch.phi_field[j] * JxW`on the left-hand side, and `scratch.old_field_values[q] * scratch.phi_field[i]` on the right-hand side, so basically solution = old_solution. But we want that to be the old_solution that already includes the reactions, otherwise our new solution doesn't make any sense. 
I updated the description to be more precise.